### PR TITLE
fix: consistently set executor log options 

### LINKED ISSUE #12788: JSON logging does not work for `resource` template
### Pre-requisites

- [X] I have double-checked my configuration
- [X] I can confirm the issue exists when I tested with `:latest`
- [X] I have searched existing issues and could not find a match for this bug
- [ ] I'd like to contribute the fix myself (see [contributing guide](https://github.com/argoproj/argo-workflows/blob/main/docs/CONTRIBUTING.md))

### What happened/what did you expect to happen?

Worfklows of type "resource" are not logging in JSON format, but output log is mixed:
```
time="2024-03-12T15:03:58.630Z" level=info msg="Starting Workflow Executor" version=v3.5.5
{"level":"info","msg":"Starting Workflow Executor","time":"2024-03-12T15:03:57.648Z","version":"v3.5.5"}
{"Duration":1000000000,"Factor":1.6,"Jitter":0.5,"Steps":5,"level":"info","msg":"Using executor retry strategy","time":"2024-03-12T15:03:57.651Z"}
{"deadline":"0001-01-01T00:00:00Z","includeScriptOutput":false,"level":"info","msg":"Executor initialized","namespace":"dev36560","podName":"hook-backend-reset-jpcwz-get-cm-940356409","templateName":"get-cm","time":"2024-03-12T15:03:57.651Z","version":"\u0026Version{Version:v3.5.5,BuildDate:2024-02-29T20:59:20Z,GitCommit:c80b2e91ebd7e7f604e88442f45ec630380effa0,GitTag:v3.5.5,GitTreeState:clean,GoVersion:go1.21.7,Compiler:gc,Platform:linux/amd64,}"}
{"level":"info","msg":"Loading manifest to /tmp/manifest.yaml","time":"2024-03-12T15:03:57.683Z"}
time="2024-03-12T15:03:58.633Z" level=info msg="Using executor retry strategy" Duration=1s Factor=1.6 Jitter=0.5 Steps=5
time="2024-03-12T15:03:58.633Z" level=info msg="Executor initialized" deadline="0001-01-01 00:00:00 +0000 UTC" includeScriptOutput=false namespace=dev36560 podName=hook-backend-reset-jpcwz-get-cm-940356409 templateName=get-cm version="&Version{Version:v3.5.5,BuildDate:2024-02-29T20:59:20Z,GitCommit:c80b2e91ebd7e7f604e88442f45ec630380effa0,GitTag:v3.5.5,GitTreeState:clean,GoVersion:go1.21.7,Compiler:gc,Platform:linux/amd64,}"
time="2024-03-12T15:03:58.645Z" level=info msg="Loading manifest to /tmp/manifest.yaml"
time="2024-03-12T15:03:58.645Z" level=info msg="kubectl get -f /tmp/manifest.yaml -o json"
time="2024-03-12T15:03:58.674Z" level=info msg="Resource: dev36560/configmap./backend-reset. SelfLink: api/v1/namespaces/dev36560/configmaps/backend-reset"
time="2024-03-12T15:03:58.674Z" level=info msg="Saving resource output parameters"
{"level":"info","msg":"Start loading input artifacts...","time":"2024-03-12T15:03:57.683Z"}
{"level":"info","msg":"Alloc=7178 TotalAlloc=12946 Sys=24165 NumGC=4 Goroutines=4","time":"2024-03-12T15:03:57.683Z"}
time="2024-03-12T15:03:58.674Z" level=info msg="kubectl -n dev36560 get configmap./backend-reset -o jsonpath={.data.wipe}"
time="2024-03-12T15:03:58.715Z" level=info msg=kubectl args="[kubectl -n dev36560 get configmap./backend-reset -o jsonpath={.data.wipe}]" error="<nil>" out=no
time="2024-03-12T15:03:58.715Z" level=info msg="Saved output parameter: wipe, value: no"
{"argo":true,"error":null,"level":"info","msg":"sub-process exited","time":"2024-03-12T15:03:59.606Z"}
```

### Version

3.5.5

### Paste a small workflow that reproduces the issue. We must be able to run the workflow; don't enter a workflows that uses private images.

```YAML
{
  "name": "get-cm",
  "inputs": {
    "parameters": [
      {
        "name": "serviceName"
      }
    ]
  },
  "outputs": {
    "parameters": [
      {
        "name": "wipe",
        "valueFrom": {
          "jsonPath": "{.data.wipe}"
        }
      }
    ]
  },
  "metadata": {},
  "resource": {
    "action": "get",
    "manifest": "apiVersion: v1\nkind: ConfigMap\nmetadata:\n    name: {{inputs.parameters.serviceName}}-reset\n"
  }
}
```

### Logs from the workflow controller

```text
{"Phase":"","ResourceVersion":"5025152597","level":"info","msg":"Processing workflow","namespace":"dev36560","time":"2024-03-12T15:03:57.082Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"warning","msg":"Non-transient error: configmaps \"artifact-repositories\" not found","time":"2024-03-12T15:03:57.090Z"}
{"artifactRepositoryRef":{"default":true},"level":"info","msg":"resolved artifact repository","time":"2024-03-12T15:03:57.090Z"}
{"level":"info","msg":"Task-result reconciliation","namespace":"dev36560","numObjs":0,"time":"2024-03-12T15:03:57.090Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"Updated phase  -\u003e Running","namespace":"dev36560","time":"2024-03-12T15:03:57.090Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"warning","msg":"Node was nil, will be initialized as type Skipped","namespace":"dev36560","time":"2024-03-12T15:03:57.090Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"was unable to obtain node for , letting display name to be nodeName","namespace":"dev36560","time":"2024-03-12T15:03:57.090Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"Steps node hook-backend-reset-jpcwz initialized Running","namespace":"dev36560","time":"2024-03-12T15:03:57.090Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"StepGroup node hook-backend-reset-jpcwz-2771728536 initialized Running","namespace":"dev36560","time":"2024-03-12T15:03:57.090Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"warning","msg":"Node was nil, will be initialized as type Skipped","namespace":"dev36560","time":"2024-03-12T15:03:57.090Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"Pod node hook-backend-reset-jpcwz-940356409 initialized Pending","namespace":"dev36560","time":"2024-03-12T15:03:57.090Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"Created pod: hook-backend-reset-jpcwz[0].get-cm (hook-backend-reset-jpcwz-get-cm-940356409)","namespace":"dev36560","time":"2024-03-12T15:03:57.125Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"Workflow step group node hook-backend-reset-jpcwz-2771728536 not yet completed","namespace":"dev36560","time":"2024-03-12T15:03:57.126Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"TaskSet Reconciliation","namespace":"dev36560","time":"2024-03-12T15:03:57.126Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"reconcileAgentPod","namespace":"dev36560","time":"2024-03-12T15:03:57.126Z","workflow":"hook-backend-reset-jpcwz"}
{"Workflow Size":22557,"level":"info","msg":"Workflow to be dehydrated","time":"2024-03-12T15:03:57.126Z"}
{"level":"info","msg":"Workflow update successful","namespace":"dev36560","phase":"Running","resourceVersion":"5025152621","time":"2024-03-12T15:03:57.147Z","workflow":"hook-backend-reset-jpcwz"}
{"Phase":"Running","ResourceVersion":"5025152621","level":"info","msg":"Processing workflow","namespace":"dev36560","time":"2024-03-12T15:04:07.126Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"Task-result reconciliation","namespace":"dev36560","numObjs":1,"time":"2024-03-12T15:04:07.126Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"task-result changed","namespace":"dev36560","nodeID":"hook-backend-reset-jpcwz-940356409","time":"2024-03-12T15:04:07.126Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"node changed","namespace":"dev36560","new.message":"","new.phase":"Succeeded","new.progress":"0/1","nodeID":"hook-backend-reset-jpcwz-940356409","old.message":"","old.phase":"Pending","old.progress":"0/1","time":"2024-03-12T15:04:07.126Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"Step group node hook-backend-reset-jpcwz-2771728536 successful","namespace":"dev36560","time":"2024-03-12T15:04:07.127Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"node hook-backend-reset-jpcwz-2771728536 phase Running -\u003e Succeeded","namespace":"dev36560","time":"2024-03-12T15:04:07.127Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"node hook-backend-reset-jpcwz-2771728536 finished: 2024-03-12 15:04:07.127389443 +0000 UTC","namespace":"dev36560","time":"2024-03-12T15:04:07.127Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"StepGroup node hook-backend-reset-jpcwz-2838691917 initialized Running","namespace":"dev36560","time":"2024-03-12T15:04:07.127Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"SG Outbound nodes of hook-backend-reset-jpcwz-940356409 are [hook-backend-reset-jpcwz-940356409]","namespace":"dev36560","time":"2024-03-12T15:04:07.127Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"Skipping hook-backend-reset-jpcwz[1].wipe-database: when 'no == yes' evaluated false","namespace":"dev36560","time":"2024-03-12T15:04:07.127Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"Skipped node hook-backend-reset-jpcwz-1176794328 initialized Skipped (message: when 'no == yes' evaluated false)","namespace":"dev36560","time":"2024-03-12T15:04:07.127Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"Skipping hook-backend-reset-jpcwz[1].wipe-redis-rabbit: when 'no == yes' evaluated false","namespace":"dev36560","time":"2024-03-12T15:04:07.127Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"Skipped node hook-backend-reset-jpcwz-3061444223 initialized Skipped (message: when 'no == yes' evaluated false)","namespace":"dev36560","time":"2024-03-12T15:04:07.127Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"Step group node hook-backend-reset-jpcwz-2838691917 successful","namespace":"dev36560","time":"2024-03-12T15:04:07.127Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"node hook-backend-reset-jpcwz-2838691917 phase Running -\u003e Succeeded","namespace":"dev36560","time":"2024-03-12T15:04:07.127Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"node hook-backend-reset-jpcwz-2838691917 finished: 2024-03-12 15:04:07.127629886 +0000 UTC","namespace":"dev36560","time":"2024-03-12T15:04:07.127Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"StepGroup node hook-backend-reset-jpcwz-3912312438 initialized Running","namespace":"dev36560","time":"2024-03-12T15:04:07.127Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"SG Outbound nodes of hook-backend-reset-jpcwz-1176794328 are [hook-backend-reset-jpcwz-1176794328]","namespace":"dev36560","time":"2024-03-12T15:04:07.127Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"SG Outbound nodes of hook-backend-reset-jpcwz-3061444223 are [hook-backend-reset-jpcwz-3061444223]","namespace":"dev36560","time":"2024-03-12T15:04:07.127Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"Skipping hook-backend-reset-jpcwz[2].wipe-elasticsearch: when 'no == yes' evaluated false","namespace":"dev36560","time":"2024-03-12T15:04:07.127Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"Skipped node hook-backend-reset-jpcwz-1313095581 initialized Skipped (message: when 'no == yes' evaluated false)","namespace":"dev36560","time":"2024-03-12T15:04:07.127Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"Step group node hook-backend-reset-jpcwz-3912312438 successful","namespace":"dev36560","time":"2024-03-12T15:04:07.127Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"node hook-backend-reset-jpcwz-3912312438 phase Running -\u003e Succeeded","namespace":"dev36560","time":"2024-03-12T15:04:07.127Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"node hook-backend-reset-jpcwz-3912312438 finished: 2024-03-12 15:04:07.12774789 +0000 UTC","namespace":"dev36560","time":"2024-03-12T15:04:07.127Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"StepGroup node hook-backend-reset-jpcwz-2771287251 initialized Running","namespace":"dev36560","time":"2024-03-12T15:04:07.127Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"SG Outbound nodes of hook-backend-reset-jpcwz-1313095581 are [hook-backend-reset-jpcwz-1313095581]","namespace":"dev36560","time":"2024-03-12T15:04:07.127Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"Skipping hook-backend-reset-jpcwz[3].update-cm: when 'no == yes' evaluated false","namespace":"dev36560","time":"2024-03-12T15:04:07.127Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"Skipped node hook-backend-reset-jpcwz-3060846225 initialized Skipped (message: when 'no == yes' evaluated false)","namespace":"dev36560","time":"2024-03-12T15:04:07.127Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"Step group node hook-backend-reset-jpcwz-2771287251 successful","namespace":"dev36560","time":"2024-03-12T15:04:07.127Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"node hook-backend-reset-jpcwz-2771287251 phase Running -\u003e Succeeded","namespace":"dev36560","time":"2024-03-12T15:04:07.127Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"node hook-backend-reset-jpcwz-2771287251 finished: 2024-03-12 15:04:07.127856308 +0000 UTC","namespace":"dev36560","time":"2024-03-12T15:04:07.127Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"Outbound nodes of hook-backend-reset-jpcwz-3060846225 is [hook-backend-reset-jpcwz-3060846225]","namespace":"dev36560","time":"2024-03-12T15:04:07.127Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"Outbound nodes of hook-backend-reset-jpcwz is [hook-backend-reset-jpcwz-3060846225]","namespace":"dev36560","time":"2024-03-12T15:04:07.127Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"node hook-backend-reset-jpcwz phase Running -\u003e Succeeded","namespace":"dev36560","time":"2024-03-12T15:04:07.127Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"node hook-backend-reset-jpcwz finished: 2024-03-12 15:04:07.127886147 +0000 UTC","namespace":"dev36560","time":"2024-03-12T15:04:07.127Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"TaskSet Reconciliation","namespace":"dev36560","time":"2024-03-12T15:04:07.127Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"reconcileAgentPod","namespace":"dev36560","time":"2024-03-12T15:04:07.127Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"Updated phase Running -\u003e Succeeded","namespace":"dev36560","time":"2024-03-12T15:04:07.127Z","workflow":"hook-backend-reset-jpcwz"}
{"level":"info","msg":"Marking workflow completed","namespace":"dev36560","time":"2024-03-12T15:04:07.127Z","workflow":"hook-backend-reset-jpcwz"}
{"Workflow Size":25319,"level":"info","msg":"Workflow to be dehydrated","time":"2024-03-12T15:04:07.128Z"}
{"action":"deletePod","key":"dev36560/hook-backend-reset-jpcwz-1340600742-agent/deletePod","level":"info","msg":"cleaning up pod","time":"2024-03-12T15:04:07.133Z"}
{"level":"info","msg":"Workflow update successful","namespace":"dev36560","phase":"Succeeded","resourceVersion":"5025154988","time":"2024-03-12T15:04:07.144Z","workflow":"hook-backend-reset-jpcwz"}
{"action":"labelPodCompleted","key":"dev36560/hook-backend-reset-jpcwz-get-cm-940356409/labelPodCompleted","level":"info","msg":"cleaning up pod","time":"2024-03-12T15:04:07.172Z"}
```

### Logs from in your workflow's wait container

```text
There is no wait container...

kubectl logs -c wait -l workflows.argoproj.io/workflow=hook-backend-reset,workflow.argoproj.io/phase!=Succeeded
error: container wait is not valid for pod hook-backend-reset-get-cm-4199754250
```

### PR DESCRIPTION
**Pull request by agilgur5**
Fixes #12788, Fixes https://github.com/argoproj/argo-workflows/pull/12929#discussion_r1572707894

### Motivation

- loglevel, gloglevel, and log format were inconsistently set around the codebase for executor containers
  - meaning that some had partial settings and some had no settings, depending on the command run etc

### Modifications

- for all commands, including `emissary`, `agent`, `resource`, and `data`
  - set loglevel, gloglevel, and log format

- replace `getExecutorLogLevel` helper with `getExecutorLogOpts` helper
- add a helper in `util/cmd/glog.go` to get the gloglevel

### Verification

Existing tests pass

### CODE REVIEW

File: workflow/controller/operator_test.go [hunk 1]
```diff
@@ -3297,11 +3297,8 @@ func TestResolveIOPathPlaceholders(t *testing.T) {
 	assert.NoError(t, err)
 	assert.True(t, len(pods.Items) > 0, "pod was not created successfully")
 
-	assert.Equal(t, []string{
-		"/var/run/argo/argoexec", "emissary",
-		"--loglevel", getExecutorLogLevel(), "--log-format", woc.controller.cliExecutorLogFormat,
-		"--", "sh", "-c", "head -n 3 <\"/inputs/text/data\" | tee \"/outputs/text/data\" | wc -l > \"/outputs/actual-lines-count/data\"",
-	}, pods.Items[0].Spec.Containers[1].Command)
+	assert.Equal(t, append(append([]string{"/var/run/argo/argoexec", "emissary"}, woc.getExecutorLogOpts()...),
+		"--", "sh", "-c", "head -n 3 <\"/inputs/text/data\" | tee \"/outputs/text/data\" | wc -l > \"/outputs/actual-lines-count/data\""), pods.Items[0].Spec.Containers[1].Command)
```
Comment (agilgur5) [inline workflow/controller/operator_test.go:3301]: huh, I thought `gofmt` would have something to say about this / auto-fix this 🤔

**Review by tczhao — COMMENTED** [review]
Shall we also comment in the CLIs
that `gloglevel` is k8s go-client logging level range 0-10

**Comment by agilgur5** [comment]
You mean making [this line](https://github.com/argoproj/argo-workflows/blob/ed55b02992d5774e6d7ccb56bef73c27cb0c5fd2/cmd/argo/commands/root.go#L138) more specific?

I didn't change the option in this PR, just made it actually used, but could improve that for sure

**Comment by agilgur5** [comment]
Merged `main` for new checks per #13027

**Comment by agilgur5** [comment]
Rebased and fixed a good chunk of merge conflicts since this has been waiting for review 6+ months now 😕 

@tczhao @terrytangyuan could one of you review this?

> You mean making [this line](https://github.com/argoproj/argo-workflows/blob/ed55b02992d5774e6d7ccb56bef73c27cb0c5fd2/cmd/argo/commands/root.go#L138) more specific?
> 
> I didn't change the option in this PR, just made it actually used, but could improve that for sure, especially as it is [for](https://github.com/kubernetes/client-go/blob/235936510e4ea63c889836f7d6b06ddcf6dfa08e/go.mod#L31) [the `klog` fork](https://github.com/kubernetes/klog)

I can also do this in a separate PR, it's a bit orthogonal to this change

**Review by tczhao — APPROVED** [review]

**Pull request merged**